### PR TITLE
tests/lib/nested.sh: split out additional helper for adding files to VM imgs

### PR DESCRIPTION
**Pull request by anonymouse64**
This is needed for testing ubuntu-seed MAAS config on UC20 nested tests, where
we need to add multiple / many files to a UC20 image file before booting it.

Also don't check cloud-init if the VM is not using cloud-init.

Broken out from https://github.com/snapcore/snapd/pull/10573

See the spread test in https://github.com/snapcore/snapd/pull/10573 for an example of how this is used.

### CODE REVIEW

File: tests/lib/nested.sh [hunk 1]
```diff
@@ -1026,8 +1032,10 @@ nested_start_core_vm_unit() {
         nested_exec "sudo snap wait system seed.loaded"
         # Copy tools to be used on tests
         nested_prepare_tools
-        # Wait for cloud init to be done
-        nested_exec "retry --wait 1 -n 5 sh -c 'cloud-init status --wait'"
+        # Wait for cloud init to be done if the system is using cloud-init
+        if [ "$NESTED_USE_CLOUD_INIT" = true ]; then
+            nested_exec "retry --wait 1 -n 5 sh -c 'cloud-init status --wait'"
+        fi
```
Comment (mardy) [inline tests/lib/nested.sh:1038]: It's not obvious why this change is here; it sounds like it could belong to a separate commit, since this is not part of the refactoring.

**Comment by sergiocazzolato** [comment]
Some tests are failing, I think a merge to master is needed

File: tests/lib/nested.sh (hunk 1 above)
Comment (anonymouse64) [reply tests/lib/nested.sh:1038]: done this is split up hopefully it's clearer now

**Review by mardy — APPROVED** [review]
LGTM!

File: tests/lib/nested.sh (hunk 1 above)
Comment (mardy) [reply tests/lib/nested.sh:1038]: It is, thanks!

**Review by sergiocazzolato — APPROVED** [review]
Nice, thanks
